src/functions.js

```diff
@@ -1,51 +1,74 @@
-}
+  return new SVGO(svgo);
+};
 
-export const getAllFileNames = (folderDirectory) => {
-    return fs.readdirSync(folderDirectory);
-}
+export const getAllFileNames = folderDirectory => {
+  return fs.readdirSync(folderDirectory);
+};
 
-export const getFileData = (fileDirectory) => {
-    return fs.readFileSync(fileDirectory);
-}
+export const getFileData = fileDirectory => {
+  return fs.readFileSync(fileDirectory);
+};
 
 export const exportSvg = (svgo, data, fileName) => {
-    svgo.optimize(data)
+  svgo
+    .optimize(data)
     .then(result => {
-        let exportDirectory = EXPORT_DIRECTORY + fileName;
+      let exportDirectory = EXPORT_DIRECTORY + fileName;
 
-        tools.ExportSVG(result.data, exportDirectory)
+      tools
+        .ExportSVG(result.data, exportDirectory)
         .then(() => console.log(`Exported: ${fileName}`))
         .catch(err => console.log(err));
     })
     .catch(err => console.log(err));
-}
+};
 
 export const buildReport = directory => {
-    console.log(`\ndirectory: ${directory}`);
-    const fileNames = getAllFileNames(directory);
-    fileNames.forEach(fileName=> {
-        const filePath = directory + fileName;
-        const stat = fs.statSync(filePath)
-        console.log(`${fileName}  ${stat.size}`)
-    })
-}
+  console.log(`\ndirectory: ${directory}`);
+  const fileNames = getAllFileNames(directory);
+  fileNames.forEach(fileName => {
+    const filePath = directory + fileName;
+    const stat = fs.statSync(filePath);
+    console.log(`${fileName}  ${stat.size}`);
+  });
+};
 
 export const comparisonReport = (originalDirectory, optimizedDirectory) => {
-    console.log(`${originalDirectory} ->  ${optimizedDirectory}`);
-    const fileNames = getAllFileNames(originalDirectory);
-    fileNames.forEach(fileName => {
-        const origionalFilePath = originalDirectory + fileName;
-        const optimizedFilePath = optimizedDirectory + fileName;
-        const origionalSize = fs.statSync(origionalFilePath).size;
-        const optimizedSize = fs.statSync(optimizedFilePath).size;
-        console.log(`${fileName} :  ${origionalSize} -> ${optimizedSize}`);
-    })
-}
+  console.log(`${originalDirectory} ->  ${optimizedDirectory}`);
+  const fileNames = getAllFileNames(originalDirectory);
+  fileNames.forEach(fileName => {
+    const originalFilePath = originalDirectory + fileName;
+    const optimizedFilePath = optimizedDirectory + fileName;
+    const originalSize = fs.statSync(origionalFilePath).size;
+    const optimizedSize = fs.statSync(optimizedFilePath).size;
+    console.log(`${fileName} :  ${origionalSize} -> ${optimizedSize}`);
+  });
+};
+
+export const writeSvgCodeIntoSvgFiles = directory => {
+  const fileNames = getAllFileNames(directory);
+  fileNames.forEach(fileName => {
+    var filePath = directory + fileName;
+    var oldData = getFileData(filePath).toString();
+    if (oldData.search(CODE_ELEMENT) < 0) {
+      var PositionToInsert = oldData.indexOf("</svg>");
```

Comment:
does this not need a +1

Reply:
if it doesnt, please send me a text snippet of an entire svg file that has been processed with your code

Reply:
Alright, I have discovered that I should add my code before </svg> no longer immediately after <svg>.

Reply:
imagine a file that start with a lot of comments or something like the following statement :```
 <?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 23.0.4, SVG Export Plug-In . SVG Version: 6.00 Build 0)  --> 
``` ,

In cant use the first ">" as there might be many ">"  before I reach the that is enclosing svg. 

There other approach will always work.

Reply:
it should be at the very top of the svg
svgomg should remove all def tags and style tags as well as all comments, so inserting it athe top should be easy

Reply:
Alright, DOne